### PR TITLE
Tuya TS0205 Smoke Detector - fix Battery Status Report

### PR DESCRIPTION
**Pull request by pixeldoc2000**
Added bind of genPowerCfg and reporting for batteryPercentageRemaining.
Fix Issue with Device goes offline and does not report Battery Status.

Related zigbee2mqtt Issue: [#19910](https://github.com/Koenkk/zigbee2mqtt/issues/19910)

### CODE REVIEW

File: src/devices/tuya.ts [hunk 1]
```diff
@@ -535,13 +535,24 @@ const definitions: Definition[] = [
...
+            {vendor: 'Dongguan Daying Electornics Technology', model: 'YG400A (Zigbee)'},
+        ],
         fromZigbee: [fz.ias_smoke_alarm_1, fz.battery, fz.ignore_basic_report],
         toZigbee: [],
+        configure: async (device, coordinatorEndpoint, logger) => {
+            try {
+                const endpoint = device.getEndpoint(1);
+                await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
+                await reporting.batteryPercentageRemaining(endpoint);
+            } catch (error) {/* Fails for some*/}
```
Comment (Koenkk) [inline src/devices/tuya.ts:554]: How did you know it fails for some?

Comment (pixeldoc2000) [reply src/devices/tuya.ts:554]: Actually, I don't know, copied it from a similar device.

Comment (Koenkk) [reply src/devices/tuya.ts:554]: Then let's leave it out.

Comment (Chartreusito) [reply src/devices/tuya.ts:554]: @pixeldoc2000 I've already submitted a [PR](https://github.com/Koenkk/zigbee2mqtt.io/pull/2406) picture for this device named "YG400A", could you remove the "(Zigbee)" from the model name ?

File: src/devices/tuya.ts [hunk 2]
```diff
@@ -529,13 +529,22 @@ const definitions: Definition[] = [
         exposes: [e.gas(), e.tamper()],
     },
     {
+        fingerprint: [{modelID: 'TS0205', manufacturerName: '_TZ3210_up3pngle'}],
```
Comment (Koenkk) [inline src/devices/tuya.ts:532]: This is not needed, `zigbeeModel: ['TS0205'],` already makes sure all gets detected as supported.

**Comment by Chartreusito** [comment]
@pixeldoc2000 could you have a look at last Koenkk suggestion ?

**Comment by emilianogetino** [comment]
Has the problem of not reporting the battery percentage been solved?

Greetings, thank you.

**Comment by skrepkaq** [comment]
@emilianogetino i tried this external converter https://github.com/Koenkk/zigbee2mqtt/issues/19910#issuecomment-1840343690, and it fixed the issue for me
image

**Comment by emilianogetino** [comment]
At the moment it seems to work fine this external converter https://github.com/Koenkk/zigbee2mqtt/issues/19910#issuecomment-1840343690

thanks

**Comment by Koenkk** [comment]
Updated and merged the pr!

**Pull request merged**